### DISCUSSION TITLE
Configuration setup for instant data saving and loading in Pyroscope

### DISCUSSION THREAD

**Discussion by susaasus1** [grafana/pyroscope#3642] [Q&A]
Hello guys! Thanks for such a great tool! I have a question: is there a way to configure Pyroscope for instant data saving so that after a reboot, it can immediately load the data once it starts? We have an issue where we run Pyroscope in Docker, and the data and data-compactor directories are set as host volumes. We have a scheduled restart every day at 3 AM, but after the restart, we lose data for about 15 minutes before the restart, meaning from 2:45 to 3:02 there is no data available after the restart. Is there a way to prevent data loss and ensure the data is loaded?"

Example:

<img width="816" alt="image" src="https://github.com/user-attachments/assets/266dca11-e9d5-43cc-ad73-2f135d37d10e">

**Comment by fvallenilla** [comment]
Did you figure out a workaround or solution here @susaasus1?
